services/mapper.go

```diff
@@ -208,140 +207,92 @@ func TraceOps(
-		to := evmClient.MustChecksum(trace.To.String())
-
-		if shouldAdd {
-			fromOp := &RosettaTypes.Operation{
-				OperationIdentifier: &RosettaTypes.OperationIdentifier{
-					Index: int64(len(ops) + startIndex),
-				},
-				Type:   traceType,
-				Status: RosettaTypes.String(opStatus),
-				Account: &RosettaTypes.AccountIdentifier{
-					Address: from,
-				},
-				Amount: &RosettaTypes.Amount{
-					Value:    new(big.Int).Neg(trace.Value).String(),
-					Currency: sdkTypes.Currency,
-				},
-				Metadata: metadata,
-			}
-			if zeroValue {
-				fromOp.Amount = nil
-			} else {
-				_, destroyed := destroyedAccounts[from]
-				if destroyed && opStatus == sdkTypes.SuccessStatus {
-					destroyedAccounts[from] = new(big.Int).Sub(destroyedAccounts[from], trace.Value)
-				}
-			}
-
-			ops = append(ops, fromOp)
+		if _, ok := destroyedAccountBalance[from]; ok && opStatus == sdkTypes.SuccessStatus {
+			destroyedAccountBalance[from] = new(big.Int).Sub(destroyedAccountBalance[from], value)
 		}
+		ops = append(ops, fromOp)
 
-		// Add to destroyed accounts if SELFDESTRUCT
-		// and overwrite existing balance.
-		if traceType == sdkTypes.SelfDestructOpType {
-			destroyedAccounts[from] = new(big.Int)
+		// Add to the destroyed account balance if SELFDESTRUCT, and overwrite existing balance.
+		if opType == sdkTypes.SelfDestructOpType {
+			destroyedAccountBalance[from] = new(big.Int)
 
-			// If destination of of SELFDESTRUCT is self,
-			// we should skip. In the EVM, the balance is reset
-			// after the balance is increased on the destination
-			// so this is a no-op.
+			// If destination of of SELFDESTRUCT is self, we should skip.
+			// In the EVM, the balance is reset after the balance is increased on the destination, so this is a no-op.
 			if from == to {
 				continue
 			}
 		}
 
-		// Skip empty to addresses (this may not
-		// actually occur but leaving it as a
-		// sanity check)
-		if len(trace.To.String()) == 0 {
-			continue
-		}
-
-		// If the account is resurrected, we remove it from
-		// the destroyed accounts map.
-		if sdkTypes.CreateType(traceType) {
-			delete(destroyedAccounts, to)
+		// If the account is resurrected, we remove it from the destroyed account balance map.
+		if sdkTypes.CreateType(opType) {
+			delete(destroyedAccountBalance, to)
 		}
 
-		if shouldAdd {
-			lastOpIndex := ops[len(ops)-1].OperationIdentifier.Index
-			toOp := &RosettaTypes.Operation{
-				OperationIdentifier: &RosettaTypes.OperationIdentifier{
-					Index: lastOpIndex + 1,
-				},
-				RelatedOperations: []*RosettaTypes.OperationIdentifier{
-					{
-						Index: lastOpIndex,
-					},
-				},
-				Type:   traceType,
-				Status: RosettaTypes.String(opStatus),
-				Account: &RosettaTypes.AccountIdentifier{
-					Address: to,
-				},
-				Amount: &RosettaTypes.Amount{
-					Value:    trace.Value.String(),
-					Currency: sdkTypes.Currency,
+		lastOpIndex := ops[len(ops)-1].OperationIdentifier.Index
+		toOp := &RosettaTypes.Operation{
```

Comment:
I suggest we make zero operation optional to include. Set by a config.
Just in case, things goes wrong, we have a mitigation.